Tests/Opc.Ua.Core.Tests/Security/Certificates/CertificateValidatorTest.cs

```diff
@@ -716,7 +723,14 @@ public async Task VerifyNotAfterInvalid(bool trusted)
             }
             var certValidator = InitValidatorWithStores();
             var serviceResultException = Assert.Throws<ServiceResultException>(() => { certValidator.Validate(cert); });
-            Assert.AreEqual(StatusCodes.BadCertificateTimeInvalid, serviceResultException.StatusCode, serviceResultException.Message);
+            if (trusted)
+            {
+                Assert.AreEqual(StatusCodes.BadCertificateUntrusted, serviceResultException.StatusCode, serviceResultException.Message);
```

Comment:
Strange, I'd expect this to be reversed? If I have a trusted cert, I would expect to get notified that the time is over, so I can suppress it. When the cert is not even trusted, I expect first to be told that it is not trusted, then I trust it and I'm told that there is yet another issue with time. Maybe the cert validator callback needs to called multiple times for app compat in such a case, until all suppressible errors are approved?